test_libs/pyspec/eth2spec/test/sanity/test_blocks.py

```diff
@@ -253,6 +253,56 @@ def test_attester_slashing(spec, state):
     )
 
 
+@with_all_phases
+@spec_state_test
+def test_proposer_after_inactive_index(spec, state):
+    # disable some low validator index to check after for
+    inactive_index = 10
+    state.validators[inactive_index].exit_epoch = spec.get_current_epoch(state)
+
+    # skip forward, get brand new proposers
+    state.slot = spec.SLOTS_PER_EPOCH * 2
+
+    while True:
+        proposer_index = spec.get_beacon_proposer_index(state)
+        if proposer_index > inactive_index:
+            # found a proposer that has a higher index than a disabled validator
+            yield 'pre', state
+            # test if the proposer can be recognized correctly after the inactive validator
+            signed_block = sign_block(spec, state, build_empty_block(spec, state), proposer_index=proposer_index)
```

Comment:
I think you want `state_transition_and_sign_block` here to make the `post` be the post state of the block transition